libfreerdp/common/assistance.c

```diff
@@ -95,21 +91,29 @@ int freerdp_assistance_crypt_derive_key_sha1(BYTE* hash, int hashLength, BYTE* k
 	buffer = (BYTE*) calloc(1, hashLength * 2);
 
 	if (!buffer)
-		return -1;
-
-	SHA1_Init(&hashCtx);
-	SHA1_Update(&hashCtx, pad1, 64);
-	SHA1_Final((void*) buffer, &hashCtx);
-
-	SHA1_Init(&hashCtx);
-	SHA1_Update(&hashCtx, pad2, 64);
-	SHA1_Final((void*) &buffer[hashLength], &hashCtx);
+		goto fail;
```

Comment:
Just a remark I think the jump here isn't necessary. A return rc would be good as well at this point.